c10/xpu/XPUMacros.h

```diff
@@ -1,15 +1,29 @@
 #pragma once
 
+#ifndef C10_USING_CUSTOM_GENERATED_MACROS
+#include <c10/xpu/impl/xpu_cmake_macros.h>
+#endif
+
 // See c10/macros/Export.h for a detailed explanation of what the function
 // of these macros are.  We need one set of macros for every separate library
 // we build.
 
+#ifdef _WIN32
+#if defined(C10_XPU_BUILD_SHARED_LIBS)
+#define C10_XPU_EXPORT __declspec(dllexport)
```

Comment:
Why can't you use the regular C10_EXPORT / C10_API here?

Reply:
No particular reason besides alignment with other backends, we can use `C10_EXPORT`. 
Same for Linux, we can use `C10_EXPORT` instead of `#define C10_XPU_EXPORT __attribute__((__visibility__("default")))`.